npm/install.js

```diff
@@ -36,23 +36,55 @@ fs.unlink(path.join(installScriptLocation, 'cmf-portal'), (err) => {
     }
 });
 
-// download respective release zip from github
-console.info(`Current platform / arch: ${process.platform} / ${process.arch}`);
-const pkgUrl = `https://github.com/criticalmanufacturing/portal-sdk/releases/download/#{Version}#/Cmf.CustomerPortal.Sdk.Console-${process.env.npm_package_version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`;// opts.binUrl.replace("{{version}}", opts.version).replace("{{platform}}", PLATFORM_MAPPING[process.platform]).replace("{{arch}}", ARCH_MAPPING[process.arch]);
-console.info(`Getting release archive from ${pkgUrl} into ${path.resolve(installScriptLocation)}`);
-axios.get(pkgUrl, { responseType: 'arraybuffer' })
-     .then(function (response) {
-         // handle success
-         const zip = tmp.tmpNameSync();
-         console.log(`Writing temporary zip file to ${zip}`);
-         fs.writeFileSync(zip, response.data);
-         console.log(`Extracting zip file ${zip} to ${installScriptLocation}`);
-         (new AdmZip(zip)).extractAllTo(installScriptLocation, undefined, true);
-         rimraf.sync(zip)
-     })
-     .catch(function (error) {
-         // handle error
-         console.error(error);
-         console.error(error(`Could not install version ${process.env.npm_package_version} on your platform ${process.platform}/${process.arch}: ${error.message}`));
-         process.exit(1);
-     });
+// Function to download and extract the zip file
+async function downloadAndExtract(pkgUrl, installScriptLocation) {
+    try {
+        console.info(`Fetching release archive from ${pkgUrl}`);
+        const response = await axios.get(pkgUrl, { responseType: 'arraybuffer' });
```

Comment:
I'm not sure if we need to maintain axios, the standard http/fetch api should be enough. Still, I'd just create a WI for later, this is not an issue

Reply:
Agreed